From 0c9035e0323a9c245607d6eaff7ef3b9f46d3d6d Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Mon, 11 May 2020 09:56:48 +0300
Subject: [PATCH] [SW-12779] preboot: skip resets for onix platforms

FLR and HOT Reset registers are not part of PSOC block.
Accessing them on Onix platforms is causing an issue
where the onix is completely getting reset. This issue
is present with both ARM and ARC scenarios. To avoid this
behavior, skip the reset checking for onix platforms
by marking it as a weak function.

Change-Id: I5052f3f2a67c9cd85ab592eec50ac3f537eb00c2
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/common/module_pci.c b/common/module_pci.c
index 0060664..688bbc2 100644
--- a/common/module_pci.c
+++ b/common/module_pci.c
@@ -29,7 +29,7 @@
 	pci_dev_specific_setup();
 }
 
-enum cmd_check_res module_pci_check(void)
+__attribute__((weak)) enum cmd_check_res dev_specific_module_pci_check(void)
 {
 	pci_sts.flr_reg_val = regmap_readl(mmPCIE_AUX_FLR_INT);
 	pci_sts.hot_rst_reg_val = regmap_readl(mmPCIE_CORE_HOT_RST) & BIT(0);
@@ -40,6 +40,11 @@
 	return NOT_FOUND;
 }
 
+enum cmd_check_res module_pci_check(void)
+{
+	return dev_specific_module_pci_check();
+}
+
 void module_pci_run(void)
 {
 	if (pci_sts.flr_reg_val == 1) {
diff --git a/goya2/onix.c b/goya2/onix.c
index 21bc4ce..30c2dfa 100644
--- a/goya2/onix.c
+++ b/goya2/onix.c
@@ -10,6 +10,7 @@
 #include <preboot.h>
 #include <pll.h>
 #include <uart.h>
+#include <cmd_module.h>
 
 #define SYS_CLOCK_FREQUENCY	35000000ULL
 
@@ -38,4 +39,14 @@
 	return (factor + 1) & 0xFFFE;
 }
 
+enum cmd_check_res dev_specific_module_pci_check(void)
+{
+	/*
+	 * FLR and Hot Reset Registers are not part of PSOC and hence
+	 * writing to them results in ERROR that is resetting the
+	 * ONIX. To avoid it, simply return with NOT FOUND
+	 */
+	return NOT_FOUND;
+}
+
 void preboot_sync_btl(void) {}
diff --git a/goya2_arc_gnu/onix.c b/goya2_arc_gnu/onix.c
index 8b11127..63e2013 100644
--- a/goya2_arc_gnu/onix.c
+++ b/goya2_arc_gnu/onix.c
@@ -10,6 +10,7 @@
 #include <preboot.h>
 #include <pll.h>
 #include <uart.h>
+#include <cmd_module.h>
 
 #include <psoc_global_conf_regs.h>
 
@@ -42,5 +43,15 @@
 	return (factor + 1) & 0xFFFE;
 }
 
+enum cmd_check_res dev_specific_module_pci_check(void)
+{
+	/*
+	 * FLR and Hot Reset Registers are not part of PSOC and hence
+	 * writing to them results in ERROR that is resetting the
+	 * ONIX. To avoid it, simply return with NOT FOUND
+	 */
+	return NOT_FOUND;
+}
+
 void preboot_init_pcie_phy(void) {}
 
